code/modules/antagonists/changeling/powers/mutations.dm

```diff
@@ -244,30 +246,14 @@
 		chain = firer.Beam(src, icon_state = "tentacle", time = INFINITY, maxdistance = INFINITY, beam_sleep_time = 1)
 	..()
 
-/obj/item/projectile/tentacle/proc/reset_throw(mob/living/carbon/human/H)
-	if(H.in_throw_mode)
-		H.throw_mode_off() //Don't annoy the changeling if he doesn't catch the item
-
-/mob/proc/tentacle_grab(mob/living/carbon/C)
-	if(Adjacent(C))
-		var/obj/item/grab/G = C.grabbedby(src,1)
-		if(istype(G))
-			G.state = GRAB_AGGRESSIVE //Instant aggressive grab
-
 /mob/proc/tentacle_stab(mob/living/carbon/C)
 	if(Adjacent(C))
 		var/obj/item/I = r_hand
-		if(!is_sharp(I))
+		if(!I || istype(/obj/item/gun/magic/tentacle, I))
 			I = l_hand
-		if(!is_sharp(I))
+		if(!I || istype(/obj/item/gun/magic/tentacle, I))
 			return
```

Comment:
re: previous comment, I misunderstood why this is like this.
It's really really cursed, but I don't know how else to do this.